### PR TITLE
feat: add go-directory inputs to ci-lint-go action

### PR DESCRIPTION
**Pull request by momentmaker**

### CODE REVIEW

File: actions/ci-lint-go/action.yml [hunk 1]
```diff
@@ -127,9 +130,9 @@ runs:
         # use-go-cache set to true
         skip-pkg-cache: ${{ inputs.use-go-cache }}
         skip-build-cache: ${{ inputs.use-go-cache }}
-
         version: ${{ inputs.golangci-lint-version }}
         args: ${{ inputs.golangci-lint-args }}
+        working-directory: ${{ inputs.go-directory }}
```
Comment (erikburt) [inline actions/ci-lint-go/action.yml:135]: Should the working directory also be applied to the shell step?

Comment (erikburt) [reply actions/ci-lint-go/action.yml:135]: What about the `upload-artifact` step?

Comment (momentmaker) [reply actions/ci-lint-go/action.yml:135]: good catch!

**Review by gheorghestrimtu — APPROVED** [review]

**Comment by momentmaker** [comment]
tested here:
https://github.com/smartcontractkit/atlas/pull/4983

**Review by erikburt — APPROVED** [review]

**Pull request merged**